### PR TITLE
python312Packages.rising: patch for python 3.12

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 339178` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.rising</li>
    <li>python311Packages.rising.dist</li>
    <li>python312Packages.rising</li>
    <li>python312Packages.rising.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 339178` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.rising</li>
    <li>python311Packages.rising.dist</li>
    <li>python312Packages.rising</li>
    <li>python312Packages.rising.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 339178` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.rising</li>
    <li>python311Packages.rising.dist</li>
    <li>python312Packages.rising</li>
    <li>python312Packages.rising.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 339178` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.rising</li>
    <li>python311Packages.rising.dist</li>
    <li>python312Packages.rising</li>
    <li>python312Packages.rising.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 339178` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.rising</li>
    <li>python311Packages.rising.dist</li>
    <li>python312Packages.rising</li>
    <li>python312Packages.rising.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 339178` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.rising</li>
    <li>python311Packages.rising.dist</li>
    <li>python312Packages.rising</li>
    <li>python312Packages.rising.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/rising/default.nix [hunk 1]
```diff
@@ -60,5 +73,7 @@ buildPythonPackage rec {
     homepage = "https://rising.rtfd.io";
     license = lib.licenses.mit;
     maintainers = with lib.maintainers; [ bcdarwin ];
+    # AttributeError: module 'configparser' has no attribute 'SafeConfigParser'. Did you mean: 'RawConfigParser'?
+    # broken = pythonAtLeast "3.12";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/rising/default.nix:77]: Does this still apply?

Comment (GaetanLepage) [reply pkgs/development/python-modules/rising/default.nix]: Nope, I forgot to remove it.
Now it is gone.

File: pkgs/development/python-modules/rising/default.nix [hunk 2]
```diff
@@ -31,6 +31,13 @@ buildPythonPackage rec {
     hash = "sha256-sBzVTst5Tp2oZZ+Xsg3M7uAMbucL6idlpYwHvib3EaY=";
   };
 
+  # Patch Python 3.12 incompatibilities in versioneer.py.
+  postPatch = ''
+    substituteInPlace versioneer.py \
+      --replace-fail SafeConfigParser ConfigParser \
+      --replace-fail readfp read_file
+  '';
+
```
Comment (NickCao) [inline pkgs/development/python-modules/rising/default.nix:40]: A better fix is to use packaged versioneer, as in `pkgs/development/python-modules/skein/default.nix`

Comment (GaetanLepage) [reply pkgs/development/python-modules/rising/default.nix]: Indeed, thanks.
I changed it.

File: pkgs/development/python-modules/rising/default.nix [hunk 3]
```diff
@@ -33,6 +34,14 @@ buildPythonPackage rec {
 
   pythonRelaxDeps = [ "lightning-utilities" ];
 
+  # Remove vendorized versioneer (incompatible with python 3.12)
+  postPatch = ''
+    substituteInPlace versioneer.py \
+      rm versioneer.py
```
Comment (NickCao) [inline pkgs/development/python-modules/rising/default.nix:40]:
```suggestion
    rm versioneer.py
```

Comment (GaetanLepage) [reply pkgs/development/python-modules/rising/default.nix]: Oops. Fixed.

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 339178` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.rising</li>
    <li>python311Packages.rising.dist</li>
    <li>python312Packages.rising</li>
    <li>python312Packages.rising.dist</li>
  </ul>
</details>

**Review by NickCao — APPROVED** [review]
Result of `nixpkgs-review pr 339178` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.rising</li>
    <li>python311Packages.rising.dist</li>
    <li>python312Packages.rising</li>
    <li>python312Packages.rising.dist</li>
  </ul>
</details>

**Pull request merged**